src/libmongoc/src/mongoc/mongoc-server-description.c

```diff
@@ -601,33 +601,33 @@ mongoc_server_description_handle_hello (mongoc_server_description_t *sd,
              * MUST treat this an authentication error." */
             sd->error.domain = MONGOC_ERROR_CLIENT;
             sd->error.code = MONGOC_ERROR_CLIENT_AUTHENTICATE;
-            goto failure;
+            goto authfailure;
```

Comment:
I forgot to put this in the ticket, but using the `GOTO` macro here would allow us to know which field failed.

```suggestion
            GOTO (authfailure);
```